src/trio/_core/_run.py

```diff
@@ -1895,7 +1886,7 @@ async def python_wrapper(orig_coro: Awaitable[RetT]) -> RetT:
                 return await orig_coro
 
             coro = python_wrapper(coro)
-        assert coro.cr_frame is not None, "Coroutine frame should exist"
+        assert coro.cr_frame is not None, "Coroutine frame should exist"  # type: ignore[attr-defined]
```

Comment:
You could add an `assert isinstance(coro, types.CoroutineType)` above this to remove this type ignore

Reply:
Hm, good point on not wanting `assert` in `_core`. I'm pretty sure this will be traced to a `nursery.start_soon`? Might be good to check this doesn't cause an internal error.

Reply:
This actually breaks on Cython, so I'm putting it back to a type ignore